lib/util/processErrors.ts

```diff
@@ -20,9 +20,12 @@ const addFileInfo = <T extends NodeError<T>>(error: T): T => {
     try {
       const titleObject: TitleObject | undefined = JSON.parse(title)
       if (titleObject !== undefined) {
+        const path = titleObject.path
```

Comment:
prefer `const { path } = titleObject`

Reply:
in fact, overall would prefer `const { path, position, url, title} = titleObject;` and then something like `const hydratedError = { ...error, path, position, url, title }`. Overall, this function both changes the error in place and returns the error object reference which can be confusing since I might expect a function which returns the error to give me a brand new error object and not edit in place (like most built in javascript functions). It would be a lot nicer to just copy the error and move forward with this if possible. Functions with side effects like this are harder to reason about

Reply:
`const hydratedError = { ...error, path, position, url, title }` I'm trying hard to switch all code to immutable and not to modify existing objects but it's not subject of this PR. This function must mutate the `error` parameter.

Reply:
sure that's fair.. should be considered a bit of tech debt

Reply:
@billytrend there's HUGE tech debt in OAV :-) after we moved to TypeScript.

Reply:
Yeah but I would say it's valuable to fix this tech debt as you make small changes to the project. EG this function could have been made side-effect free in this PR while you had decent context on what it was doing and in a big change later

Reply:
@billytrend too much risk for this bug fix. This function (relaxEntryType) is the first in my list:

https://github.com/Azure/oav/pull/339/files#diff-8f10dca8bc7f28f14841b4aa28b52a8eL711